src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1654,7 +1657,7 @@ AvailableExtensionVersion(void)
 			Datum availableVersion = slot_getattr(tupleTableSlot, 2, &isNull);
 
 			/* we will cache the result of citus version to prevent catalog access */
-			oldMemoryContext = MemoryContextSwitchTo(CacheMemoryContext);
+			oldMemoryContext = MemoryContextSwitchTo(MetadataCacheMemoryContext);
```

Comment:
We might be using this one before `MetadataCacheMemoryContext` is initialised because it is called from CitusHasBeenLoaded().

Reply:
InitializeCaches() is called earlier in the function